### PR TITLE
Correctly interpret call context with DotExprs.

### PR DESCRIPTION
**Pull request by varomodt**
This upstreams an internal change which corrects the Buildifier's
rewrite behavior when the call site is a DotExpr as opposed to an Ident.

### CODE REVIEW

File: build/rewrite.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ import (
 	"sort"
 	"strings"
 
-	"github.com/bazelbuild/buildtools/tables"
```
Comment (laurentlb) [inline build/rewrite.go]: please revert this change

File: build/rewrite.go [hunk 2]
```diff
@@ -460,12 +467,21 @@ func sortStringLists(f *File, info *RewriteInfo) {
...
+				// this context is in the blacklist.
+				if disabled("unsafesort") || !f.Build || tables.SortableBlacklist[context] {
 					continue
 				}
-				if disabled("unsafesort") && !tables.SortableWhitelist[context] && !allowedSort(context) {
+
+				// Drop out unless this field name is whitelisted, we're explicitly
+				// allowed to sort it for debugging reasons, or its field name is
+				// explicitly allowed.
+				if !tables.SortableWhitelist[context] && !allowedSort(context) && !tables.IsSortableListArg[key.Name] {
```
Comment (laurentlb) [inline build/rewrite.go]: For clarification: is there any behavior change here?

File: build/rewrite.go (hunk 1 above)
Comment (varomodt) [reply build/rewrite.go]: Done. Sorry about that -- I'm not used to github.

File: build/rewrite.go (hunk 2 above)
Comment (varomodt) [reply build/rewrite.go]: There are some minor behavior changes here:

- If "unsafesort" is disabled, then this makes this rewrite _never_ apply, whereas at head it will still apply if allowedSort returns true or the field is in SortableWhitelist, which seemed inconsistent with the semantics implied by the "disabled" method. Note that every other case here already drops out immediately if unsafesort is disabled.

- This makes SortableWhitelist and allowedSort allow a field to be sorted even if the given key.Name is not in IsSortableListArg: the current behavior seems to violate the implicit semantics of an explicit SortableWhitelist.
